messaging/src/main/java/org/axonframework/eventhandling/AbstractEventBus.java

```diff
@@ -123,31 +133,38 @@ public void publish(@Nonnull List<? extends EventMessage<?>> events) {
                          () -> "It is not allowed to publish events when the root Unit of Work has already been " +
                                  "committed.");
 
-            unitOfWork.afterCommit(u -> ingested.forEach(MessageMonitor.MonitorCallback::reportSuccess));
-            unitOfWork.onRollback(uow -> ingested.forEach(
-                    message -> message.reportFailure(uow.getExecutionResult().getExceptionResult())
-            ));
+            unitOfWork.afterCommit(u -> {
+                ingested.forEach(MessageMonitor.MonitorCallback::reportSuccess);
+                span.end();
```

Comment:
So, a form of `spans.forEach(Span::end)` instead of a single span.
Or, we do keep them grouped, as in essence we want to publish everything in one go instead of separately on this level.

Reply:
I'd propose to check what it looks like.